src/Hpack/Config.hs

```diff
@@ -440,18 +466,27 @@ data Section a = Section {
 , sectionInstallIncludes :: [FilePath]
 , sectionLdOptions :: [LdOption]
 , sectionBuildable :: Maybe Bool
-, sectionConditionals :: [Conditional]
+, sectionConditionals :: [Conditional b]
 , sectionBuildTools :: Dependencies
-} deriving (Eq, Show, Functor, Foldable, Traversable)
+} deriving (Show)
```

Comment:
I noticed that in the parsed result we actually don't need two type parameters (we only have `Section Library Library` and `Section Executable Executable`).  The second type parameter is only needed during parsing.

As a second observation: We have types that represent the syntactical structure of `package.yaml` that we use for parsing the actual YAML.  We then map those syntactic types to types that represent a `Package` semantically (e.g. `ExecutableConfig` is mapped to `Executable`).

For `Section` the syntactic type is `CommonOptions`, but here we break the strict separation between syntactic and semantic types.  We use `Section` during parsing the actual YAML as well.

What I did is: Remove the `FromJSON` instance for `Section` and only use `CommonOptions` during parsing.